src/components/learner-credit-management/requests-tab/data/hooks/useBnrSubsidyRequests.js

```diff
@@ -0,0 +1,176 @@
+import {
+  useCallback, useMemo, useState,
+} from 'react';
+import { logError } from '@edx/frontend-platform/logging';
+import { camelCaseObject } from '@edx/frontend-platform/utils';
+import { debounce } from 'lodash-es';
+import EnterpriseAccessApiService from '../../../../../data/services/EnterpriseAccessApiService';
+import { REQUEST_TAB_VISIBLE_STATES } from '../constants';
+
+// Map table column accessors to API field names for sorting
+const API_FIELDS_BY_TABLE_COLUMN_ACCESSOR = {
+  requestDetails: 'requestDetails',
+  amount: 'amount',
+  requestDate: 'requestDate',
+  requestStatus: 'requestStatus',
+};
+
+// Hardcoded filter choices for request status
+// This is a temporary solution until the backend supports dynamic filtering
```

Comment:
[Question] do we have a ticket to handle the work to support dynamic filtering?

Reply:
Yes. https://2u-internal.atlassian.net/browse/ENT-10345